Assignment7.txt

```diff
@@ -0,0 +1,22 @@
+//Anya Lauria 
+
+1. ++*p increments the variable that the pointer is pointing to first, and then evaluates it, similar to prefix 
+   incrementing in regular variables. *p++ evaluates the variable that the  pointer is pointing to first, then 
+   advances it. Not sure about *++p 
+2. The left to right order and right to left order depends on the kind of operation. For example, post fix operations 
+   always take precedence. The order can be manipulated using parantheses. 
+3. You can use them in function arguments, and not copy large chunks of memory, so the use of pointers is efficient. 
+   Pointers are also used to return arrays- there is no other way. Instead of using a character array, we can use 
+   pointers to work with strings instead, which is much easier. They also allow us to dynamic memory allocation, and
+   resize data structure whenever needed. 
+4.  4.1 char * 
+    4.2 invalid 
```

Comment:
This is actually valid! "xyz" is just an array of characters, so the "xyz"[1] is just accessing "y". Then you just subtract 'y' from it to get 0. This is because, if you can recall, a char is just a value mapped to a character!